src/Database.hs

```diff
@@ -33,12 +34,17 @@ getConnection = do
 
 getConnectInfos :: IO ConnectInfo
 getConnectInfos = do
-  host <- getEnv "PG_HOST"
-  port <- getEnv "PG_PORT"
-  user <- getEnv "PG_USER"
-  pass <- getEnv "PG_PASS"
-  dbname <- getEnv "PG_DBNAME"
-  pure $ ConnectInfo host (read port) user pass dbname
+  host <- lookupEnv "PG_HOST"
+  port <- lookupEnv "PG_PORT"
+  user <- lookupEnv "PG_USER"
+  pass <- lookupEnv "PG_PASS"
+  dbname <- lookupEnv "PG_DBNAME"
+
+  pure $ ConnectInfo (getConnectInfo host connectHost) (getConnectInfo port connectPort) (getConnectInfo user connectUser) (getConnectInfo pass connectPassword) (getConnectInfo dbname connectDatabase)
+
+getConnectInfo :: Read t => Maybe String -> (ConnectInfo -> t) -> t
+getConnectInfo (Just info) _ = read info
```

Comment:
We should be carefull with the `read` function, the instance for `String` expect an enquoted string literal (will only be able to parse `"\"hello\""` and not `"hello"`).
I bet that you expected the instance of `read` for `String` to be the identity, but it isn't :).